hudi-common/src/main/java/org/apache/hudi/common/table/TableSchemaResolver.java

```diff
@@ -130,7 +131,11 @@ private MessageType getTableParquetSchemaFromDataFile() throws Exception {
                     + " for file " + filePathWithFormat.getLeft());
             }
           } else {
-            return readSchemaFromLastCompaction(lastCompactionCommit);
+            if (lastCompactionCommit.isPresent()) {
```

Comment:
how about using `lastCompactionCommit.map().orElseThrow()`. instead of the if-else